cmd/vaas-hook/main.go

```diff
@@ -83,9 +104,66 @@ func getCommonFlags() []cli.Flag {
 			Destination: &Config.Debug,
 			EnvVar:      "VAAS_HOOK_DEBUG",
 		},
+		cli.StringFlag{
+			Name:        action.FlagVaaSURL,
+			Usage:       "address of the VaaS endpoint",
+			Destination: &Config.VaaSURL,
+			EnvVar:      "VAAS_URL",
+		},
+		cli.StringFlag{
+			Name:        action.FlagUser,
+			Usage:       "user for Auth",
+			Destination: &Config.VaaSUser,
+			EnvVar:      "VAAS_USER",
+		},
+		cli.StringFlag{
+			Name:        action.FlagSecretKey,
+			Usage:       "client key for Auth",
+			Destination: &Config.VaaSKey,
+			EnvVar:      "VAAS_KEY",
+		},
+		cli.StringFlag{
+			Name:        action.FlagDirector,
+			Usage:       "VaaS director to register this backend with",
+			Destination: &Config.Director,
+			Value:       Config.Director,
+		},
+		cli.StringFlag{
+			Name:        action.FlagAddress,
+			Usage:       "IP address of this backend",
+			Destination: &Config.Address,
+			Value:       Config.Address,
+		},
+		cli.IntFlag{
+			Name:        action.FlagPort,
+			Usage:       "port of this backend",
+			Destination: &Config.Port,
+			Value:       Config.Port,
+		},
+		cli.BoolFlag{
+			Name:  action.FlagDryRun,
+			Usage: "do not perform any changes, just print requests",
+		},
+		cli.BoolFlag{
+			Name:  action.FlagCanaryTag,
+			Usage: "this backend is a canary",
+		},
 	}
 }
 
 func getCommands() []cli.Command {
-	return []cli.Command{}
+	return []cli.Command{
```

Comment:
Please be consistent and create the same cli interface as proposed in other already existing hook: https://github.com/allegro/consul-registration-hook/blob/master/cmd/consul-registration-hook/main.go#L22

in this case:
 ```
./vaas-registration-hook k8s register
./vaas-registration-hook k8s deregister
```

Reply:
Ok, this will take a bit of refactoring and will address the `k8s.GetPodInfo()` issue too.